src/circuit/circuit_frame.rs

```diff
@@ -3075,6 +3439,108 @@ fn apply_continuation<F: LurkField, CS: ConstraintSystem<F>>(
     Ok((result_expr, result_env, result_cont, make_thunk_num))
 }
 
+fn hide<F: LurkField, CS: ConstraintSystem<F>>(
+    mut cs: CS,
+    secret: F,
+    maybe_payload: &AllocatedPtr<F>,
+    store: &Store<F>,
+) -> Result<AllocatedPtr<F>, SynthesisError> {
+    let hide_ptr = if let Some(ptr) = maybe_payload.ptr(store).as_ref() {
+        match store.hidden(secret, *ptr) {
+            Some(c) => c,
+            None => store.get_nil(),
+        }
+    } else {
+        store.get_nil()
```

Comment:
I think this is a real 'dummy' value — and that it will never be used because `maybe_payload` is only absent from the store when the reduction being proved is not indeed a hiding operation. If so, let's mark it as such with a comment.

Also, have we standardized on a value to use in such cases? NIL is okay, but it could easily be confused (when reading code or debugging) with a real value. Maybe we should allocate something with a fabricated tag and access it like `store.get_bogus_dummy()` or something like that — killing two birds with one stone (i.e. also documenting intent). Probably there are other places such a value should go, but we could start with creating it here and propagate over time (if needed).